src/Atc.Azure.Options/Extensions/EnvironmentOptionsExtensions.cs

```diff
@@ -1,13 +1,17 @@
-﻿using Atc.Azure.Options.Environment;
+using Atc.Azure.Options.Environment;
 
 namespace Atc.Azure.Options.Extensions
 {
     public static class EnvironmentOptionsExtensions
     {
-        public static string GetKeyVault(this EnvironmentOptions options)
-            => $"https://{options.GetResourceName()}.vault.azure.net";
+        public static string GetKeyVault(this EnvironmentOptions options, NamingOptions namingOptions)
+            => $"https://{options.GetResourceName(namingOptions)}.vault.azure.net";
 
-        public static string GetResourceName(this EnvironmentOptions options)
-            => string.Concat(options.CompanyAbbreviation, options.SystemAbbreviation, options.EnvironmentName, options.ServiceAbbreviation);
+        public static string GetResourceName(this EnvironmentOptions options, NamingOptions namingOptions)
```

Comment:
Consider how we can extend this to support different naming schemes. This could be with a NamingProvider where we provide a default implementation that is our standard matching the snippets project.